src/vmm_mad/remotes/lib/vcenter_driver/virtual_machine.rb

```diff
@@ -1092,6 +1200,32 @@ def set_boot_order(boot_info)
         RbVmomi::VIM.VirtualMachineBootOptions({bootOrder: boot_order})
     end
 
+    def sync_nics(option = :none, execute = true)
+        device_change = []
+
+        if option == :all
+            nics_each(:detached?) do |nic|
+                device_change << {
+                    :operation => :remove,
+                    :device    => nic.vc_item
+                }
+            end
+        end
+
+        nics_each(:no_exists?) do |nic|
+            device_change << calculate_add_nic_spec(nic.one_item)
+        end
+
+        return device_change unless execute
+
+        spec_hash = { :deviceChange => device_change }
+
+        spec = RbVmomi::VIM.VirtualMachineConfigSpec(spec_hash)
+        @item.ReconfigVM_Task(:spec => spec).wait_for_completion
+
+        info_nics
+    end
+
     # TODO
     # Synchronize the OpenNebula VM representation with vCenter VM
```

Comment:
Briefly explain what this does with NICs and DISKs, if they a) exist in both vcenter and ONE, b) exist only in vcenter, c) exist only in ONE